registry/registry-nacos/src/main/java/com/alipay/sofa/rpc/registry/nacos/NacosRegistry.java

```diff
@@ -272,25 +286,28 @@ public List<ProviderGroup> subscribe(final ConsumerConfig config) {
             }
 
             try {
-                if (providerObserver == null) {
-                    providerObserver = new NacosRegistryProviderObserver();
+                ProviderInfoListener providerInfoListener = config.getProviderInfoListener();
+                try {
+                    lock.lock();
```

Comment:
better to leave `lock()` outside try block where `unlock()` appears.
ref to https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/locks/ReentrantLock.html

There are some discussion on this topic https://stackoverflow.com/questions/10868423/lock-lock-before-try